src/schedule.rs

```diff
@@ -478,6 +483,16 @@ mod test {
         assert_eq!(prev, next);
     }
 
+    #[test]
+    fn test_next_after_past_date_next_year() {
+        let past = Utc::now() - chrono::Duration::weeks(3);
```

Comment:
This bug only appears when using `Utc::now()`? I wonder if you could have this test use fixed dates